### PR TITLE
Remove configurable run/work dir from global config

### LINKED ISSUE #4034: remove run and work dir configurations
Remove:

* `global.cylc[platforms][<platform name>]run directory`
* `global.cylc[platforms][<platform name>]work directory`

These have been made obsolete by the `global.cylc[symlink dirs]` configurations.

(We now install all flows into `~/cylc-run/<flow-name>`, creating symlinks to locations on other filesystems where `$HOME` is not suitable).

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by MetRonnie**
These changes close #4034 

Remove from `global.cylc`:
- `[platforms][<platform name>]run directory`
- `[platforms][<platform name>]work directory`
- `[platforms][<platform name>]suite definition directory`

Also remove obsolete `--template` option from `cylc suite-state`

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (no need to test removed stuff)
- [x] Appropriate change log entry included.
- [x] (master branch) I have opened a documentation PR at cylc/cylc-doc/pull/239.
- [x] No dependency changes.

---

- [x] Check these are given deprecation notations in the docs:
  >  (Note that we also need to deprecate the environment variables `$CYLC_SUITE_DEF_PATH` & `$CYLC_SUITE_DEF_PATH_ON_SUITE_HOST`.)

### CODE REVIEW

**Comment by MetRonnie** [comment]
Not sure if there's any test cases I should implement?

**Comment by hjoliver** [comment]
> Not sure if there's any test cases I should implement?

Presumably sufficient to remove any code associated with the obsolete config items, and adapt or remove tests that use it.

**Comment by wxtim** [comment]
Testing locally I get a single failure - `tests/functional/cylc-message/00-ssh.t`. It's possible, even likely that this is caused by issues with my `global-tests.cylc`; can you check that you can run it?

**Review by wxtim — APPROVED** [review]
- [x] Tests pass locally.
- [x] Changes look sensible.

**Comment by MetRonnie** [comment]
> - [ ] Check these are given deprecation notations in the docs:
>  >  (Note that we also need to deprecate the environment variables `$CYLC_SUITE_DEF_PATH` & `$CYLC_SUITE_DEF_PATH_ON_SUITE_HOST`.)

@dpmatthews I think you brought this up - I can't find any reference to them in the docs at all, so is this point resolved, or should I add something somwhere?

**Comment by dpmatthews** [comment]
> @dpmatthews I think you brought this up - I can't find any reference to them in the docs at all, so is this point resolved, or should I add something somwhere?

We don't appear to document other deprecated env variables so I think that's fine

File: cylc/flow/job_file.py [hunk 1]
```diff
@@ -201,14 +198,7 @@ def _write_suite_environment(self, handle, job_conf, run_d):
...
         handle.write('\n')
         # override and write task-host-specific suite variables
-        work_d = get_remote_suite_work_dir(
-            job_conf["platform"], job_conf['suite_name'])
         handle.write('\n    export CYLC_SUITE_RUN_DIR="%s"' % run_d)
-        if work_d != run_d:
-            # Note: not an environment variable, but used by job.sh
-            handle.write(
-                '\n    export CYLC_SUITE_WORK_DIR_ROOT="%s"' % work_d
-            )
```
Comment (oliver-sanders) [inline cylc/flow/job_file.py:211]: This is only set in one place now (nice) so can remove the Bash var substitution in `job.sh`:

```diff
-CYLC_SUITE_WORK_DIR_ROOT="${CYLC_SUITE_WORK_DIR_ROOT:-${CYLC_SUITE_RUN_DIR}}"
+CYLC_SUITE_WORK_DIR_ROOT="${CYLC_SUITE_RUN_DIR}"
```

**Review by oliver-sanders — APPROVED** [review]

File: cylc/flow/job_file.py (hunk 1 above)
Comment (MetRonnie) [reply cylc/flow/job_file.py:211]: Seeing as that var is not exported, I've removed it altogether

**Comment by hjoliver** [comment]
Merge unit test conflict resolution, and confirmed that the result passed.

**Pull request merged**